### PR TITLE
Return an error when ranking.queryCache is not enabled when it needs …

### PR DESCRIPTION
**Pull request by baldersheim**
…to be.
@bratseth PR This is a followup from #3120 
@havardpe FYI

### CODE REVIEW

**Comment by bratseth** [comment]
👍

File: container-search/src/main/java/com/yahoo/prelude/fastsearch/VespaBackEndSearcher.java [hunk 1]
```diff
@@ -109,6 +112,35 @@
 
     protected abstract void doPartialFill(Result result, String summaryClass);
 
+    protected static boolean wantsRPCSummarFill(Query query) {
```
Comment (havardpe) [inline container-search/src/main/java/com/yahoo/prelude/fastsearch/VespaBackEndSearcher.java]: Summar -> Summary

File: container-search/src/main/java/com/yahoo/prelude/fastsearch/VespaBackEndSearcher.java [hunk 2]
```diff
@@ -109,6 +112,35 @@
...
+        if (docsumDefinition == null) return true; // stay safe
+        if (docsumDefinition.isDynamic()) return true;
+
+        // Needed to generate ranking features?
+        RankProfile rankProfile = documentDb.rankProfiles().get(query.getRanking().getProfile());
+        if (rankProfile == null) return true; // stay safe
+        if (rankProfile.hasSummaryFeatures()) return true;
+        if (query.getRanking().getListFeatures()) return true;
+
+        // (Don't just add other checks here as there is a return false above)
```
Comment (havardpe) [inline container-search/src/main/java/com/yahoo/prelude/fastsearch/VespaBackEndSearcher.java]: This comment makes no sense. If there are more reasons to need the query in the future, where do I put those checks?

Comment (bratseth) [inline container-search/src/main/java/com/yahoo/prelude/fastsearch/VespaBackEndSearcher.java]: I think the intended meaning is "don't just add other checks here *without thinking*"

Comment (havardpe) [inline container-search/src/main/java/com/yahoo/prelude/fastsearch/VespaBackEndSearcher.java]: That could be said about any function. Also, I think the return false earlier implies less thinking, not more. I was just curious about exactly which checks the author was afraid someone would add.

File: container-search/src/main/java/com/yahoo/prelude/fastsearch/VespaBackEndSearcher.java (hunk 1 above)
Comment (baldersheim) [inline container-search/src/main/java/com/yahoo/prelude/fastsearch/VespaBackEndSearcher.java]: Fixed in https://github.com/vespa-engine/vespa/pull/3140

**Pull request merged**